src/components/periodic-report/periodic-report.service.ts

```diff
@@ -0,0 +1,306 @@
+      id,
+      userId: session.userId,
+    });
+    if (!id) {
+      throw new NotFoundException(
+        'No periodic report id to search for',
+        'periodicReport.id'
+      );
+    }
+
+    const query = this.db
+      .query()
+      .call(matchRequestingUser, session)
+      .match([node('node', 'PeriodicReport', { id })])
+      .call(matchPropList)
+      .optionalMatch([
+        node('node'),
+        relation('out', '', 'reportFile', { active: true }),
+        node('reportFile', 'File'),
+      ])
+      .return('node, propList, reportFile.id as reportFileId')
+      .asResult<
+        StandardReadResult<DbPropsOfDto<PeriodicReport>> & {
+          reportFileId: string;
+        }
+      >();
+
+    const result = await query.first();
+    if (!result) {
+      throw new NotFoundException(
+        'Could not find periodic report',
+        'periodicReport.id'
+      );
+    }
+
+    const props = parsePropList(result.propList);
+    const securedProps = await this.authorizationService.secureProperties(
+      IPeriodicReport,
+      props,
+      session
+    );
+
+    return {
+      ...parseBaseNodeProperties(result.node),
+      ...securedProps,
+      type: props.type,
+      start: props.start,
+      end: props.end,
+      reportFile: await this.files.resolveDefinedFile(
+        {
+          value: result.reportFileId,
+          canEdit: true,
+          canRead: true,
+        },
+        session
+      ),
+      canDelete: await this.db.checkDeletePermission(id, session),
+    };
+  }
+
+  async update(
+    { reportFile, ...input }: UpdatePeriodicReport,
+    session: Session
+  ): Promise<PeriodicReport> {
+    const object = await this.readOne(input.id, session);
+
+    return await this.db.sgUpdateProperties({
+      session,
+      object,
+      props: ['start', 'end'],
+      changes: input,
+      nodevar: 'periodicReport',
+    });
+  }
+
+  async delete(id: string, session: Session): Promise<void> {
+    const object = await this.readOne(id, session);
+
+    if (!object) {
+      throw new NotFoundException(
+        'Could not find periodic report',
+        'periodicReport.id'
+      );
+    }
+
+    try {
+      await this.db.deleteNode({
+        session,
+        object,
+        aclEditProp: 'canDeleteOwnUser',
+      });
+    } catch (exception) {
+      this.logger.warning('Failed to delete periodic report', {
+        exception,
+      });
+
+      throw new ServerException('Failed to delete periodic report', exception);
+    }
+  }
+
+  async listProjectReports(
+    projectId: string,
+    reportType: ReportType,
+    { filter, ...input }: PeriodicReportListInput,
+    session: Session
+  ): Promise<SecuredPeriodicReportList> {
+    const query = this.db
+      .query()
+      .match([
+        node('project', 'Project', { id: projectId }),
+        relation('out', '', 'report', { active: true }),
+        node('node', `PeriodicReport:${reportType}Report`),
```

Comment:
Prefer just using an array.
```suggestion
        node('node', ['PeriodicReport', `${reportType}Report`]),
```